eclair-core/src/test/scala/fr/acinq/eclair/payment/PostRestartHtlcCleanerSpec.scala

```diff
@@ -543,6 +544,99 @@ class PostRestartHtlcCleanerSpec extends TestKitBaseClass with FixtureAnyFunSuit
+    val relayedhtlc1Out = buildHtlcOut(0L, channelId_ab_2, trampolineRelayedPaymentHash)
+    val nonRelayedHtlc2In = buildHtlcIn(1L, channelId_ab_1, relayedPaymentHash)
+
+    val pluginParams = new CustomCommitmentsPlugin {
+      def name = "test with incoming HTLC from remote"
+      def getIncomingHtlcs: Seq[PostRestartHtlcCleaner.IncomingHtlc] = List(PostRestartHtlcCleaner.IncomingHtlc(relayedHtlc1In.add, None), PostRestartHtlcCleaner.IncomingHtlc(nonRelayedHtlc2In.add, None))
+      def getHtlcsRelayedOut(htlcsIn: Seq[PostRestartHtlcCleaner.IncomingHtlc]): Map[Origin, Set[(ByteVector32, Long)]] = Map.empty
+    }
+
+    val nodeParams1 = nodeParams.copy(pluginParams = List(pluginParams))
+    val c = ChannelCodecsSpec.makeChannelDataNormal(List(relayedhtlc1Out), Map(0L -> trampolineRelayed))
+    nodeParams1.db.channels.addOrUpdateChannel(c)
+
+    val channel = TestProbe()
+    f.createRelayer(nodeParams1)
+    register.expectNoMsg(100 millis) // nothing should happen while channels are still offline.
+
+    val cs = new AbstractCommitments {
+      def getOutgoingHtlcCrossSigned(htlcId: Long): Option[UpdateAddHtlc] = None
+      def getIncomingHtlcCrossSigned(htlcId: Long): Option[UpdateAddHtlc] = Some(relayedHtlc1In.add)
+      def timedOutOutgoingHtlcs(blockheight: Long): Set[UpdateAddHtlc] = Set.empty
+      def localNodeId: PublicKey = randomExtendedPrivateKey.publicKey
+      def remoteNodeId: PublicKey = randomExtendedPrivateKey.publicKey
+      def capacity: Satoshi = Long.MaxValue.sat
+      def availableBalanceForReceive: MilliSatoshi = Long.MaxValue.msat
+      def availableBalanceForSend: MilliSatoshi = 0.msat
+      def originChannels: Map[Long, Origin] = Map.empty
+      def channelId: ByteVector32 = channelId_ab_1
+      def announceChannel: Boolean = false
+    }
+
+    // Non-standard channel goes to NORMAL state:
+    system.eventStream.publish(ChannelStateChanged(channel.ref, channelId_ab_1, system.deadLetters, a, OFFLINE, NORMAL, Some(cs)))
+    channel.expectMsg(CMD_FAIL_HTLC(1L, Right(TemporaryNodeFailure), commit = true))
+    channel.expectNoMsg(100 millis)
+  }
+
+  test("Relayed standard->nonstandard HTLC is retained") { f =>
+    import f._
+
+    val relayedPaymentHash = randomBytes32
+    val trampolineRelayedPaymentHash = randomBytes32
+    val trampolineRelayed = Origin.TrampolineRelayedCold((channelId_ab_2, 0L) :: Nil)
+    val relayedHtlcIn = buildHtlcIn(0L, channelId_ab_2, trampolineRelayedPaymentHash)
+    val nonRelayedHtlcIn = buildHtlcIn(1L, channelId_ab_2, relayedPaymentHash)
+
+    val pluginParams = new CustomCommitmentsPlugin {
+      def name = "test with outgoing HTLC to remote"
+      def getIncomingHtlcs: Seq[PostRestartHtlcCleaner.IncomingHtlc] = List.empty
+      def getHtlcsRelayedOut(htlcsIn: Seq[PostRestartHtlcCleaner.IncomingHtlc]): Map[Origin, Set[(ByteVector32, Long)]] = Map(trampolineRelayed -> Set((channelId_ab_2, 0L)))
```

Comment:
It doesn't make sense that htlc #0 in `channel_ab_2` has been relayed as htlc #0 in `channel_ab_2`...